### PR TITLE
Editorial: Add explicit wait step

### PR DESCRIPTION
**Pull request by syg**

### CODE REVIEW

**Comment by syg** [comment]
PTAL @bakkot and @michaelficarra. I can't figure out how to assign reviewers.

File: spec/index.html [hunk 1]
```diff
@@ -14,6 +14,7 @@ <h1>Atomics.pause ( [ _iterationNumber_ ] )</h1>
     1. If _iterationNumber_ is not *undefined*, then
       1. If _iterationNumber_ is not an integral Number, throw a *TypeError* exception.
       1. If ℝ(_iterationNumber_) &lt; 0, throw a *RangeError* exception.
+    1. If an ECMAScript implementation is running on a processor that supports an instruction to signal that the current executing code is in a spin-wait loop, it may perform that instruction.
```
Comment (bakkot) [inline spec/index.html]: I would be a little stronger and more general maybe?

```suggestion
    1. If the execution environment of the ECMAScript implementation supports a signal that the current executing code is in a spin-wait loop, for example via a hardware instruction, the implementation should send that signal.
```

Comment (ljharb) [reply spec/index.html]: s/should/must, then?

Comment (michaelficarra) [reply spec/index.html]: We don't usually should/must. Why not "... instruction, send that signal"?

**Review by bakkot — APPROVED** [review]

**Comment by michaelficarra** [comment]
The new step LGTM, though I also think we should have some normative reference to `iterationNumber`.

**Comment by syg** [comment]
> The new step LGTM, though I also think we should have some normative reference to iterationNumber.

How about this sentence that encapsulates the note from below:

"An ECMAScript implementation may send that signal multiple times, determined by _iterationNumber_ when not **undefined**. If so, the number of times the signal is sent for an integral Number _N_ is at most the number of times it is sent for _N_ + 1."

Edit: wordsmithed to be two sentences instead.

**Comment by michaelficarra** [comment]
"An ECMAScript implementation may send that signal more than once, ...". Otherwise sounds good.

**Comment by syg** [comment]
What is the difference between "more than once" and "multiple times"?

**Comment by michaelficarra** [comment]
You know, now that I think about it, either is fine. My initial reading of "multiple times" was "≥1" and I wanted it to say ">1". Don't know why I initially read it that way.

File: spec/index.html [hunk 2]
```diff
@@ -14,6 +14,7 @@ <h1>Atomics.pause ( [ _iterationNumber_ ] )</h1>
     1. If _iterationNumber_ is not *undefined*, then
       1. If _iterationNumber_ is not an integral Number, throw a *TypeError* exception.
       1. If ℝ(_iterationNumber_) &lt; 0, throw a *RangeError* exception.
+    1. If the execution environment of the ECMAScript implementation supports a signal that the current executing code is in a spin-wait loop, send that signal. An ECMAScript implementation may send that signal multiple times, determined by _iterationNumber_ when not *undefined*. The number of times the signal is sent for an integral Number _N_ is at most the number of times it is sent for _N_ + 1.
```
Comment (michaelficarra) [inline spec/index.html]:
```suggestion
    1. [declared="N"] If the execution environment of the ECMAScript implementation supports a signal that the current executing code is in a spin-wait loop, send that signal. An ECMAScript implementation may send that signal multiple times, determined by _iterationNumber_ when not *undefined*. The number of times the signal is sent for an integral Number _N_ is at most the number of times it is sent for _N_ + 1.
```

alternatively, you can just name `iterationNumber`:

```suggestion
    1. If the execution environment of the ECMAScript implementation supports a signal that the current executing code is in a spin-wait loop, send that signal. An ECMAScript implementation may send that signal multiple times, determined by _iterationNumber_ when not *undefined*. The number of times the signal is sent is at most the number of times it is sent for any integral Number greater than _iterationNumber_.
```

**Pull request merged**